Makefile

```diff
@@ -0,0 +1,22 @@
+.PHONY: test format lint coverage clean run
```

Comment:
ooo what's `.PHONY` do?

Reply:
It declares these as "phony" targets, which means they don't actually create anything.   Without this, if you have a file called, say, `test` in your code, and you run `make test`, make will see that `test` is there, and assume its job is done.

By declaring it a phony target, you tell make "this doesn't actually produce an output file, you should always run it."  It's not strictly necessary, but it can avoid unintentional confusion down the line.